tests/serverpod_test_server/test_e2e/service_protocol_test.dart

```diff
@@ -465,6 +465,23 @@ void main() {
       });
     });
   });
+
+  group('File retrieval', () {
+    test('Fetch generated/protocol.yaml file', () async {
+      var file =
+          await serviceClient.insights.fetchFile('generated/protocol.yaml');
+      expect(file, isNotNull);
+      expect(file.length, greaterThan(0));
+    });
+    test('Fetch file outside whitelist', () async {
```

Comment:
Suggestion: Add a test for the scenario when the file is not found.

There is a last assert we can throw from this endpoint that is not tested right now: `FileNotFoundException`

Reply:
I'm skipping this for now.